### PR TITLE
Riscv stable CI

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description

Exploring whether this is possible, this seems to work locally but lets try it in a CI run. Builds on top of #1485 as this removes the nightly features from the examples.

### CODE REVIEW

**Comment by MabezDev** [comment]
So it's possible, but what are the complications if we do?

**Comment by jessebraham** [comment]
Could you rebase this when you get a chance, please?

**Comment by MabezDev** [comment]
Hmm, this got a bit harder now we have the 802154 driver in tree, I guess I need to figure out what we can do with `c_variadic`

File: esp-hal/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `time::current_time` API (#1503)
 - ESP32-S3: Add LCD_CAM Camera driver (#1483)
 - `embassy-usb` support (#1517)
+- esp8021504: add `binary-logs` nightly only feature (#1491)
```
Comment (jessebraham) [inline esp-hal/CHANGELOG.md:20]: `esp-hal/CHANGELOG.md` is only for changes to the `esp-hal` package.

**Review by jessebraham — APPROVED** [review]
This doesn't need a `CHANGELOG` entry but otherwise I think it LGTM, thanks for this.

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**